prov/rxm/src/rxm_ep.c

```diff
@@ -468,6 +470,19 @@ static int rxm_getname(fid_t fid, void *addr, size_t *addrlen)
 	return fi_getname(&rxm_ep->msg_pep->fid, addr, addrlen);
 }
 
+static int rxm_join(struct fid_ep *ep, const void *addr, uint64_t flags,
+			struct fid_mc **mc, void *context)
+{
+	if (flags & FI_COLLECTIVE) {
+		struct fi_collective_addr *c_addr =
+			(struct fi_collective_addr *)addr;
+		return ofi_join_collective(ep, c_addr->coll_addr, c_addr->set,
+					   flags, mc, context);
+	} else {
+		return -FI_ENOSYS;
+	}
```

Comment:
nit: move flag check to top of function and abort.  Outdent the rest of the code.